homeassistant/components/moscow_yandex_transport/sensor.py

```diff
@@ -0,0 +1,123 @@
+ATTRIBUTION = "Data provided by maps.yandex.ru"
+
+CONF_STOP_ID = "stop_id"
+CONF_ROUTE = "routes"
+
+DEFAULT_NAME = "Yandex Transport"
+ICON = "mdi:bus"
+
+SCAN_INTERVAL = timedelta(minutes=1)
+
+PLATFORM_SCHEMA = PLATFORM_SCHEMA.extend(
+    {
+        vol.Required(CONF_STOP_ID): cv.string,
+        vol.Optional(CONF_NAME, default=DEFAULT_NAME): cv.string,
+        vol.Optional(CONF_ROUTE, default=[]): vol.All(cv.ensure_list, [cv.string]),
+
+    }
+)
+
+
+def setup_platform(hass, config, add_entities, discovery_info=None):
+    """Set up the Yandex transport sensor."""
+    stop_id = config[CONF_STOP_ID]
+    name = config[CONF_NAME]
+    routes = config[CONF_ROUTE]
+
+    data = YandexMapsRequester(user_agent=USER_AGENT)
+    add_entities([DiscoverMoscowYandexTransport(data, stop_id, routes, name)], True)
+
+
+class DiscoverMoscowYandexTransport(Entity):
+    def __init__(self, requester, stop_id, routes, name):
+        """Initialize sensor."""
+        self.requester = requester
+        self._stop_id = stop_id
+        self._routes = []
+        self._routes = routes
+        self._state = None
+        self._name = name
+        self._attrs = None
+
+    def update(self):
+        """Get the latest data from maps.yandex.ru and update the states."""
+        attrs = {}
+        closer_time = None
+        try:
+            yandex_reply = self.requester.get_stop_info(self._stop_id)
+            data = yandex_reply["data"]
+            stop_metadata = data["properties"]["StopMetaData"]
+        except KeyError as e:
+            _LOGGER.warning(
+                "Exception KeyError was captured, missing key is %s. Yandex returned: %s" % e, yandex_reply)
+            self.requester.set_new_session()
+            data = self.requester.get_stop_info(self._stop_id)["data"]
+            stop_metadata = data["properties"]["StopMetaData"]
+        stop_name = data["properties"]["name"]
+        transport_list = stop_metadata["Transport"]
+        for transport in transport_list:
+            route = transport["name"]
+            if self._routes and route not in self._routes:
+                # skip unnecessary route info
+                continue
+            if "Events" in transport["BriefSchedule"]:
+                for event in transport["BriefSchedule"]["Events"]:
+                    if "Estimated" in event:
+                        posix_time_next = int(event["Estimated"]["value"])
+                        if closer_time is None or closer_time > posix_time_next:
+                            closer_time = posix_time_next
+                        if route not in attrs:
+                            attrs[route] = []
+                        attrs[route].append(event["Estimated"]["text"])
+        attrs[STOP_NAME] = stop_name
+        attrs[ATTR_ATTRIBUTION] = ATTRIBUTION
+        if closer_time is None:
+            self._state = None
+        else:
+            self._state = datetime.utcfromtimestamp(closer_time).isoformat(timespec='minutes')
```

Comment:
right?

Reply:
Looks good. I haven't seen us test the timespec option yet, so please test that a lovelace card handles that format correctly.

Try the different options here:
https://www.home-assistant.io/lovelace/entities/#format

Reply:
i see some problem.
in last version (with min) we saw timechart:
![minutes](https://user-images.githubusercontent.com/3110317/64093250-51890700-cd60-11e9-9ae8-c61a4d978696.png)
in new version we can't look at chart:
![timestamp](https://user-images.githubusercontent.com/3110317/64093251-52219d80-cd60-11e9-98d4-68598d248a21.png)

Reply:
That's a frontend decision/problem.

Did you check that the timestamp is shown as relative time in the lovelace entities card?

Reply:
i 

> That's a frontend decision/problem.
> 
> Did you check that the timestamp is shown as relative time in the lovelace entities card?

sorry, i cant solve problem with present timestamp in relative mode:
![some_errors](https://user-images.githubusercontent.com/3110317/64298962-04837b80-cf80-11e9-985e-1d90ebe027d8.png)
can you help?

Reply:
It should be something like this:
```yaml
entities:
  - entity: sensor.yandex_transport
    format: relative
type: entities
```

Reply:
Yes, it works:
`title: Home Assistant
views:
  - badges:
      - person.rishatik92
      - sensor.yandex_transport
      - sun.sun
      - sensor.ya
    cards:
      - entity: weather.home_assistant
        type: weather-forecast
      - entities:
          - entity: sensor.yandex_transport
            format: relative
        type: entities
    path: default_view
    title: Home
  - badges:
      - sensor.yandex_transport
    cards: []
`
![it_works](https://user-images.githubusercontent.com/3110317/64381465-ee3af580-d03b-11e9-9f03-9ff5a7d160bc.png)
Yes. it works!

Reply:
Great!